docs/python/developer.qmd

```diff
@@ -0,0 +1,80 @@
+---
+title: "For developers"
+---
+
+# Set up the developer environment
+
+## Setting up conda
+
+In order to develop on `ribasim` locally, please follow the following steps:
+
+- Download and install [miniconda](https://docs.conda.io/en/latest/miniconda.html).
+
+- Initialize `conda` by running the following in the `Miniconda prompt`:
+
+```
+conda init
+```
+
+- Depending on your company settings, you might also have to run the following in a Powershell terminal as administrator:
+
+```
+Set-ExecutionPolicy -ExecutionPolicy RemoteSigned
+```
+
+## Creating (or updating) the environment
+
+- Create (or update) the environment by executing the following in your terminal:
+
+```
+conda env create --file=environment.yml --force
+```
+
+## Installing Ribasim Python
+
+- Activate the environment
+
+```
+conda activate ribasim
+```
+
+In order to develop on `ribasim` locally, execute the following line inside your virtual environment
+
+```bash
+pip install -e .
+```
+
+## Running the tests
+
+
+If no already done, activate the environment
+
+```
+conda activate ribasim
+```
+
+
+To run the tests execute:
+
+```bash
+pytest tests
+```
+
+## Setup Visual Studio Code (optional)
+
+1. Install the [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python), [ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) and [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) extensions.
+
+2. Add the following to your `.vscode/settings.json` file in your workspace
+
+```json
+{
+    "[python]": {
+        "editor.formatOnSave": true,
+        "editor.codeActionsOnSave": {
+            "source.fixAll": true
+        }
+    },
+    "python.formatting.provider": "black",
+    "autoDocstring.docstringFormat": "numpy"
+}
+```
```

Comment:
Shall we check these in, `git add -f .vscode/settings.json`? With also `"julia.environmentPath": "core"` I'd say, since the root of the repo is no longer a Julia environment.

Reply:
That way the settings.json gets tracked, which means that also unrelated changes by developers get easily included.
I now added a template instead.

Reply:
Right, I didn't think of that. Too bad that is not possible, a template file seems like a good alternative.